src/app/nice.py

```diff
@@ -0,0 +1,53 @@
+# -----------------------------------------
+# Nice and dry Phantom App
+# -----------------------------------------
+# TODO: This SHOULD be split into its own repo and imported
+import inspect
+
+# Phantom App imports
+import phantom.app as phantom
+from phantom.base_connector import BaseConnector
+
+
+def handle(*action_ids):
+    """Registor function to handle given action_ids."""
+    def decorator(func):
+        func._handle = action_ids
+        return func
+    return decorator
+
+
+class NiceBaseConnector(BaseConnector):
+
+    def __init__(self):
+        super(NiceBaseConnector, self).__init__()
+
+        self.actions = {}
+        for _, method in inspect.getmembers(self):
+            if hasattr(method, '_handle'):
+                for action_id in getattr(method, '_handle'):
+                    self.actions[action_id] = method
+
+        self._state = None
+
+    def handle_action(self, param):
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id", self.get_action_identifier())
+
+        if action_id in self.actions.keys():
+            return self.actions[action_id](param)
+
+        return phantom.APP_SUCCESS  # TODO: Should this be an error instead?
```

Comment:
Yes. I think this should be an error. "Unrecognized action was called. Did you add your action to `actions`?"

Reply:
I left it as success because that is how it was coded before, but it seemed odd.

Reply:
This is resolved - in the other repo. - See https://github.com/techservicesillinois/phantom-toolbox/blob/5ec5a1a2d3574a140208e291c15dc0350b64ec4d/src/phtoolbox/app/base_connector.py#L40